crates/uv-pep508/src/marker/algebra.rs

```diff
@@ -328,6 +333,11 @@ impl InternerGuard<'_> {
 
         let (x, y) = (self.shared.node(xi), self.shared.node(yi));
 
+        // Determine whether the conjunction _could_ contain a conflict. As an optimization, we only
+        // have to perform this check at the top-level, since these variables are given higher
+        // priority in the tree. In other words, if they're present, they _must_ be at the top.
+        let conflicts = x.var.conflicts() && y.var.conflicts();
```

Comment:
The idea is to give the "possibly-conflicting" markers highest priority, so they're always at the top of the tree.

As such, when we AND two expressions, we only have to perform the "possibly-conflicting" check if they both have a conflicting variable at the very top.